prometheus/src/main/java/org/opensearch/sql/prometheus/response/PrometheusResponse.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.prometheus.response;
+
+import static org.opensearch.sql.prometheus.data.constants.PrometheusFieldConstants.METRIC;
+import static org.opensearch.sql.prometheus.data.constants.PrometheusFieldConstants.TIMESTAMP;
+import static org.opensearch.sql.prometheus.data.constants.PrometheusFieldConstants.VALUE;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import lombok.NonNull;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.opensearch.sql.data.model.ExprDoubleValue;
+import org.opensearch.sql.data.model.ExprStringValue;
+import org.opensearch.sql.data.model.ExprTimestampValue;
+import org.opensearch.sql.data.model.ExprTupleValue;
+import org.opensearch.sql.data.model.ExprValue;
+
+public class PrometheusResponse implements Iterable<ExprValue> {
+
+  private final JSONObject responseObject;
+
+  public PrometheusResponse(JSONObject responseObject) {
+    this.responseObject = responseObject;
+  }
+
+  @NonNull
+  @Override
+  public Iterator<ExprValue> iterator() {
+    List<ExprValue> result = new ArrayList<>();
+    if ("matrix".equals(responseObject.getString("resultType"))) {
+      JSONArray itemArray = responseObject.getJSONArray("result");
+      for (int i = 0; i < itemArray.length(); i++) {
+        JSONObject item = itemArray.getJSONObject(i);
+        JSONObject metric = item.getJSONObject("metric");
+        JSONArray values = item.getJSONArray("values");
+        for (int j = 0; j < values.length(); j++) {
+          LinkedHashMap<String, ExprValue> linkedHashMap = new LinkedHashMap<>();
+          JSONArray val = values.getJSONArray(j);
+          linkedHashMap.put(TIMESTAMP,
+              new ExprTimestampValue(Instant.ofEpochMilli((long) (val.getDouble(0) * 1000))));
```

Comment:
prometheus timestamp returned is in seconds? possible to get more precise?

Reply:
Its in double format in milliseconds
1435781430.781